Lickability/PinpointKit
# Added GUI alert on mailer failure.

Closes #219.

## What It Does
Previously, if the user clicked `Send` and the `MailSender` experienced a failure (e.g. when there are no mail accounts set up), nothing would happen in the GUI. There was also no method in `PinpointKitDelegate` to correspond to `didFailToSend` in `SenderDelegate`.

This commit adds an alert popup to `PinpointKit` to solve the first problem, and a `didFailToSend` method to `PinpointKitDelegate` for the second. The latter takes the original `Error` as a parameter in case developers want to use it present more specific messages.

![simulator](https://cloud.githubusercontent.com/assets/13933679/26026468/56b2fc78-382e-11e7-816a-dfcba0abcc52.png)

## How to Test
Ensure that the mailer will encounter an error (the easiest way is to run the example app on a simulator with no email accounts set up) and click `Send`.

## Notes
None in particular.
